cranelift/codegen/src/regalloc/context.rs

```diff
@@ -209,10 +209,18 @@ impl Context {
             emit_stackmaps(func, domtree, &self.liveness, &mut self.tracker, isa);
         } else {
             // Make sure no references are used.
-            for val in func.dfg.values() {
-                let ty = func.dfg.value_type(val);
-                if ty.lane_type().is_ref() {
-                    panic!("reference types were found but safepoints were not enabled.");
+            for block in func.layout.blocks() {
```

Comment:
What about using references as function parameters, but not using them anywhere? This would skip them.

Reply:
Good point! I think a few tests actually need to be added:

- for function parameters
- for EBB params
- for value types in signature refs

Can you add the code and tests for this, please? Thanks!